crates/ndc-query-plan/src/plan_for_query_request/plan_for_grouping.rs

```diff
@@ -0,0 +1,245 @@
+                plan::Type::array_of(object_field.r#type)
+            } else {
+                object_field.r#type
+            };
+
+            plan::Dimension::Column {
+                path: relationship_path,
+                column_name,
+                arguments: plan_arguments,
+                field_path,
+                field_type,
+            }
+        }
+    };
+    Ok(plan_dimension)
+}
+
+fn plan_for_group_expression<T: QueryContext>(
+    plan_state: &mut QueryPlanState<T>,
+    object_type: &plan::ObjectType<T::ScalarType>,
+    expression: ndc::GroupExpression,
+) -> Result<plan::GroupExpression<T>> {
+    match expression {
+        ndc::GroupExpression::And { expressions } => Ok(plan::GroupExpression::And {
+            expressions: expressions
+                .into_iter()
+                .map(|expr| plan_for_group_expression(plan_state, object_type, expr))
+                .collect::<Result<_>>()?,
+        }),
+        ndc::GroupExpression::Or { expressions } => Ok(plan::GroupExpression::Or {
+            expressions: expressions
+                .into_iter()
+                .map(|expr| plan_for_group_expression(plan_state, object_type, expr))
+                .collect::<Result<_>>()?,
+        }),
+        ndc::GroupExpression::Not { expression } => Ok(plan::GroupExpression::Not {
+            expression: Box::new(plan_for_group_expression(
+                plan_state,
+                object_type,
+                *expression,
+            )?),
+        }),
+        ndc::GroupExpression::UnaryComparisonOperator { target, operator } => {
+            Ok(plan::GroupExpression::UnaryComparisonOperator {
+                target: plan_for_group_comparison_target(plan_state, object_type, target)?,
+                operator,
+            })
+        }
+        ndc::GroupExpression::BinaryComparisonOperator {
+            target,
+            operator,
+            value,
+        } => {
+            let target = plan_for_group_comparison_target(plan_state, object_type, target)?;
+            let (operator, operator_definition) = plan_state
+                .context
+                .find_comparison_operator(&target.result_type(), &operator)?;
+            let value_type = operator_definition.argument_type(&target.result_type());
+            Ok(plan::GroupExpression::BinaryComparisonOperator {
+                target,
+                operator,
+                value: plan_for_group_comparison_value(plan_state, value_type, value)?,
+            })
+        }
+    }
+}
+
+fn plan_for_group_comparison_target<T: QueryContext>(
+    plan_state: &mut QueryPlanState<T>,
+    object_type: &plan::ObjectType<T::ScalarType>,
+    target: ndc::GroupComparisonTarget,
+) -> Result<plan::GroupComparisonTarget<T>> {
+    let plan_target = match target {
+        // TODO: Do we expect the target aggregate to correspond to one of the grouping aggregates?
+        // Or can it be independent?
```

Comment:
This seems like a question? Do we know the answer and can remove this?

Reply:
Good catch - I found the answer but forgot to remove the comment